File: arrays/rotate/rotate.ts
export class Solution {
    /**
     * @param {string} testInput
     * @return {any} Returns either boolean to validate the test cases or a custom result.
     */
    run_solution(testInput: any) {
        let [nums, k] = testInput;
        k = k % nums.length;

        if (k != 0) {
            const temp = nums.slice(-k).concat(nums.slice(0, k + 1));
            nums = temp;
        }
        return nums;
    }
}

File: arrays/rotate/rotate.py
"""Solution Module"""


class Solution:
    """Solution Class"""

    def run_solution(self, test_input):
        """Solution Method"""
        [nums, k] = test_input
        print(nums[-k:])
        nums[:k], nums[k:] = nums[-k:], nums[:-k]
        return nums
